pennylane_lightning/src/tests/Makefile

```diff
@@ -24,13 +24,16 @@ all: test
 lightning_unittest.o: lightning_unittest.cpp
 	$(CC) $^ $(CFLAGS) -c
 
-lightning_util_unittest.o: lightning_util_unittest.cpp
+Gates.o: ../rework/Gates.cpp
+	$(CC) $^ $(CFLAGS) -c
+
+lightning_gates_unittest.o: lightning_gates_unittest.cpp
 	$(CC) $^ $(CFLAGS) -c
 
 gtest_main.o: gtest_main.cpp
 	$(CC) $^ $(CFLAGS) -c
 
-cpptests: gtest_main.o lightning_unittest.o lightning_util_unittest.o
```

Comment:
I'm confused about `Gates.o` :thinking:

Reply:
We need to compile (`Gates.o: ../rework/Gates.cpp` option) and link (here adding `Gates.o`) the `.cpp` sources that we depend on in our testing. In this PR it's `Gates.cpp`.

Previously, we could get away with not doing this in the case of `lightning_unittest.cpp` because tests in there only depended on definitions that were available in `.hpp` files. For example,  ​the `lightning_quibt.cpp` file only contained the definition and binding of the `apply` function: a utility function that was a switch. All other function definitions were contained in `lightning_quibt.hpp`, so we didn't need to separately compile & link this file or any other sources.

This is now different with the new backend.

Reply:
Thanks for figuring that out :) Glad I'm not the only one who got confused by that.